specs/protocol/holocene/derivation.md

```diff
@@ -1,13 +1,332 @@
+_backwards-invalidate_ previous valid batches that came from the same span batch, as pre-Holocene.
+
+An implementation may choose to put the current span batch, from which singular batches are derived
+from, into a separate stage, or process it within the batch queue under the new rules.
+
+When a batch derived from the current staging channel is a singular batch, it is directly forwarded
+to the batch queue. Otherwise, it is set as the current span batch in the span batch stage. The
+following span batch validity checks are done, before singular batches are derived from it.
+Definitions are borrowed from the [original Span Batch specs](../delta/span-batches.md).
+
+- If the span batch _L1 origin check_ is not part of the canonical L1 chain, the span batch is
+invalid.
+- A failed parent check invalidates the span batch.
+- If `span_start.timestamp > next_timestamp`, the span batch is invalid, because we disallow gaps
+due to the new strict batch ordering rules.
+- If `span_end.timestamp < next_timestamp`, the span batch is invalid, as it doesn't contain any new
+batches (this would also happen if applying timestamp checks to each derived singular batch
+individually).
+- Note that we still allow span batches to overlap with the safe chain (`span_start.timestamp < next_timestamp`).
+
+If any of the above checks invalidate the span batch, it is dropped and the remaining channel from
+which the span batch was derived, is also immediately dropped (see also
+[Fast Channel Invalidation](#fast-channel-invalidation)).
+
+## Batch Queue
+
+The batch queue is also simplified in that batches are required to arrive strictly ordered, and any
+batches that violate the ordering requirements are immediately dropped, instead of buffered.
+
+So the following changes are made to the [Bedrock Batch Queue](../derivation.md#batch-queue):
+
+- The reordering step is removed, so that later checks will drop batches that are not sequential.
+- The `future` batch validity status is removed, and batches that were determined to be in the
+future are now directly `drop`-ped. This effectively disallows gaps, instead of buffering future batches.
+- The other rules stay the same, including empty batch generation when the sequencing window
+elapses.
+
+If a batch is found to be invalid and is dropped, the remaining span batch it originated from, if
+applicable, is also discarded.
+
+Note that when the L1 origin of the batch queue moves forward, it is guaranteed that it is empty,
+because future batches aren't buffered any more.
+
+### Fast Channel Invalidation
+
+Furthermore, upon finding an invalid batch, the remaining channel it got derived from is also discarded.
+
+_TODO: I believe that the batch queue, similarly to the channel bank, now actually only holds at most
+one single staging batch, because we eagerly derive payloads from any valid singular batch. And the
+span batch stage before it would similarly only hold at most one staging span batch._
```

Comment:
We can update this after getting more familiar with the implementation of Holocene derivation

Reply:
Definitely. We may also enforce it in the implementation. I'll leave the comment in and update later during implementation.